File: Sample Delegate/Sample Delegate/ViewController.m
//
//  ViewController.m
//  Sample Delegate
//
//  Created by Nancy on 27/10/2020.
//

#import "ViewController.h"
#import "SampleProtocol.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UILabel *label;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    SampleProtocol *sampleProtocol = [[SampleProtocol alloc]init];
       sampleProtocol.delegate = self;
    [ _label setText:@"Processing..."];
       [sampleProtocol startSampleProcess];
       // Do any additional setup after loading the view, typically from a nib.
    // Do any additional setup after loading the view.
}
#pragma mark - Sample protocol delegate
-(void)processCompleted {
   [ _label setText: @"Process Completed"];
}

@end

File: outlet and actions/outlet and actions/ViewController.swift
//
//  ViewController.swift
//  outlet and actions
//
//  Created by Nancy on 20/10/2020.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
    }

    @IBOutlet weak var label1: UILabel!
    
    @IBAction func changeLabel1(_ sender: Any) {
        
        label1.text = "label 1 changed "
    }
    @IBOutlet weak var label2: UILabel!
    @IBAction func changeLabel2(_ sender: Any) {
        label2.text = "label 2 changed "
    }
}

